astronomer/airflow/version_check/update_checks.py

```diff
@@ -230,25 +232,35 @@ def parse_version(rel):
             ver = version.parse(release['version'])
             if release['channel'] in ['alpha', 'beta']:  # ignore alpha & beta releases
                 continue
-            if ver <= current_version:
+            if ver < current_version:
                 self.log.debug(
-                    "Got to a release (%s) that is older than the running version (%s) -- stopping looking for more",
+                    "Got to a release (%s) that is older or equal to the running version (%s) -- stopping looking "
+                    "for more",
                     ver,
                     self.runtime_version,
                 )
                 break
+            hidden_from_ui = ver <= current_version
 
             if 'release_date' in release:
                 release_date = pendulum.parse(release['release_date'], timezone='UTC')
             else:
                 release_date = utcnow()
 
+            end_of_support = (
+                pendulum.parse(release.get('end_of_support'), timezone='UTC')
+                if release.get('end_of_support') is not None
+                else None
+            )
+
             yield AstronomerAvailableVersion(
                 version=release['version'],
                 level=release['level'],
                 date_released=release_date,
                 url=release.get('url'),
                 description=release.get('description'),
+                end_of_support=end_of_support,
+                hidden_from_ui=hidden_from_ui,
```

Comment:
We might want to just set this for the current version, otherwise we likely will overwrite the flag if the user has already hidden it?